### PR TITLE
Use Enum.Parse<TEnum>

### PR DESCRIPTION
**Pull request by lbargaoanu**
I guess the diff is useless :) I could open another PR with the C# 10 stuff.

### CODE REVIEW

**Comment by orty** [comment]
Hi,
I just wanted to check, is this PR about the ability to use `projectTo<>()` along with `string <-> Enum` mapping ?

Many thanks for your great work on this library !

File: src/AutoMapper/Configuration/ConfigurationValidator.cs [hunk 1]
```diff
@@ -6,34 +6,21 @@
 using AutoMapper.Internal.Mappers;
 namespace AutoMapper.Configuration
 {
-    using Validator = Action<ValidationContext>;
     [EditorBrowsable(EditorBrowsableState.Never)]
-    public class ConfigurationValidator
+    public readonly record struct ConfigurationValidator(IGlobalConfigurationExpression Expression)
```
Comment (lbargaoanu) [inline src/AutoMapper/Configuration/ConfigurationValidator.cs:10]: I've abused records for the convenient syntax, but other than some extra IL at compile time, I don't see much of a downside :) I guess I want a primary constructor.

**Comment by jbogard** [comment]
I've got a request from Microsoft to push out this one: https://github.com/AutoMapper/AutoMapper/pull/3999, is this one close to wrapping?

**Comment by lbargaoanu** [comment]
Sure.

**Review by jbogard — APPROVED** [review]

**Pull request merged**